tests/testthat/test-distribution-normal.R

```diff
@@ -0,0 +1,45 @@
+#' Note: consider PyTorch - like test schema
+#' See: https://github.com/pytorch/pytorch/blob/master/test/distributions/test_distributions.py
+#' TODO: add more unit tests
+
+test_that("Distribution Normal - basic size test", {
+  
+  loc         <- torch_randn(5, 5, requires_grad=TRUE)
+  scale       <- torch_randn(5, 5)$abs()$requires_grad_()
+  loc_1d      <- torch_randn(1, requires_grad=TRUE)
+  scale_1d    <- torch_randn(1)$abs()$requires_grad_()
+  
+  expect_equal(distr_normal(loc, scale)$sample()$size(), c(5, 5))
+  expect_equal(distr_normal(loc, scale)$sample(7)$size(), c(7, 5, 5))
+  expect_equal(distr_normal(loc_1d, scale_1d)$sample(1)$size(), c(1, 1))
+  expect_equal(distr_normal(loc_1d, scale_1d)$sample()$size(), 1)
+  # expect_equal(distr_normal(0.2, .6)$sample(1)$size(), 1)
+  # expect_equal(distr_normal(-0.7, 50.0)$sample()$size(), 1)
```

Comment:
I'd add tests making sure the constraints are used and respected. A single `expect_error` would be enough.

Reply:
Yes, you're right. I'll add some tests.